### PR TITLE
Add field of regard and reachable set

### PR DESCRIPTION
**Pull request by DK-sr**
* also added simple protobuf message recorder

### CODE REVIEW

File: .gitmodules [hunk 1]
```diff
@@ -1,6 +1,3 @@
 [submodule "external/imgui"]
 	path = samples/imgui_display/imgui
 	url = https://github.com/ocornut/imgui.git
-[submodule "external/websocketpp"]
```
Comment (KangJH) [inline .gitmodules:4]: why did you remove this?

File: include/sensr_message_listener.h [hunk 2]
```diff
@@ -23,7 +23,7 @@ namespace sensr {
     MessageListener(ListeningType listening_type = ListeningType::kOutputMessage);
     virtual ~MessageListener() = default;
     virtual void OnError(Error error, const std::string& reason);
-    virtual void OnGetOutpuMessage(const sensr_proto::OutputMessage &message);
+    virtual void OnGetOutputMessage(const sensr_proto::OutputMessage &message);
```
Comment (KangJH) [inline include/sensr_message_listener.h:26]: Jerone mentioned to mark the old API as deprecated even it is miss-spelled.
I gonna handle this in the other PR.

File: proto/sensr_proto/point_cloud.proto [hunk 3]
```diff
@@ -7,11 +7,11 @@ message PointResult {
     enum Type {
       NONE = 0;
       GROUND = 1;
-      ENVIRONMENT = 2;
+      BACKGROUND = 2;
       RAW = 3;
     }
     Type type = 1;
-    string id = 2; // In calibration : algo_uri#topic, In runtime : meaningless
```
Comment (KangJH) [inline proto/sensr_proto/point_cloud.proto:14]: Please keep this comment.

File: proto/sensr_proto/output.proto [hunk 4]
```diff
@@ -78,10 +78,16 @@ message EventMessage {
   SystemHealth health = 3;
 }
 
+message CustomMessage {
```
Comment (KangJH) [inline proto/sensr_proto/output.proto:81]: Please make this in a different file for easy maintaining.

File: proto/sensr_proto/type.proto [hunk 5]
```diff
@@ -63,3 +63,12 @@ message Object {
   History history = 102;
   Prediction prediction = 103;
 }
+
+message PredictedReachableSet {
```
Comment (KangJH) [inline proto/sensr_proto/type.proto:67]: AFAIK, J made Prediction to share this info too.
Could you integrate PredictedReachableSet into Prediction.

**Review by KangJH — CHANGES_REQUESTED** [review]

File: .gitmodules (hunk 1 above)
Comment (DK-sr) [reply .gitmodules:4]: Oh I accidently remove this while removing ptcl. I will rollback this.

File: proto/sensr_proto/output.proto (hunk 4 above)
Comment (DK-sr) [reply proto/sensr_proto/output.proto:81]: Done!

**Review by jae1e — APPROVED** [review]

**Review by KangJH — APPROVED** [review]
